src/test/java/com/amihaiemil/eoyaml/AllYamlLinesTest.java

```diff
@@ -298,4 +335,80 @@ public void complainsWhenCannotTurn() {
             );
         }
     }
+
+    /**
+     * Unit test for serializing quoted keys of scalar values.
+     * @throws IOException When there's a problem reading the sample files.
+     */
+    @Test
+    public void quotedKeysRoundTripTests() throws IOException {
+        quotedKeysTest("quotedKeysMin.yml", 1);
+        quotedKeysTest("quotedKeysMax.yml", 6);
+    }
+
+    /**
+     * Do a round-trip test on the given quoted-key sample file.
+     * @param filename Sample file name. Contain a mapping keyed with
+     *                 'a_mapping'
+     * @param mapSize The size of the mapping
+     * @throws IOException When there's a problem reading the sample files.
+     */
+    private void quotedKeysTest(final String filename, final int mapSize)
+            throws IOException {
+        final List<YamlLine> fileLines = readTestResource(filename);
+
+        final String fileContents =
+                fileLines.stream().map(YamlLine::toString)
+                        .collect(Collectors.joining("\n"));
+
+        final AllYamlLines allYamlLines = new AllYamlLines(fileLines);
```

Comment:
@zsandoz These 2 tests should actually be in class ``RtYamlInputTest``, I'm almost sure. 

What you're doing here is emulating the behaviour of YamlInput... you're reading the file contents manually, parsing them into lines, then instantiate and use AllYamlLines. That's what YamlInput is doing. 

So these tests should be inside RtYamlInputTest and instantiating the YamlMapping should look like:

```java
YamlMapping read = new RtYamlInput(
    new FileInputStream(
        new File("src/test/resources/quotedKeysMax.yml")
    )
).readYamlMapping();
```

Then just make your assertions on the YamlMapping.

Reply:
Done